src/plugins/plugin-bluebird.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+var shimmer = require('shimmer');
+var is = require('is');
+
+function wrapFunction(api, fn) {
+  return fn && is.fn(fn) ? api.wrap(fn) : fn;
+}
+
+function patchPromise(Promise, api) {
+  shimmer.wrap(Promise.prototype, '_addCallbacks', function(original) {
```

Comment:
If this is all that is required for knex we don't necessarily need to add more to this PR but if we end up patching bluebird long term we may want to patch it more exhaustively. It would be good to give bluebird the same context propagation semantics as [async-listener gives to the global promise](https://github.com/othiym23/async-listener/blob/master/index.js#L337).